File: test_vpc_migration.py
#!/usr/bin/env python3
"""
Simple test script to verify VPC functionality migration works correctly.
"""

import os
from agentbay import AgentBay, CreateSessionParams

def test_vpc_session_creation():
    """Test VPC session creation and tool listing."""
    # Get API key from environment
    api_key = os.getenv('AGENTBAY_API_KEY')
    if not api_key:
        print("Error: AGENTBAY_API_KEY environment variable not set")
        return False
    
    try:
        # Initialize AgentBay client
        config = {
            'region_id': 'cn-hangzhou',
            'endpoint': 'wuyingai-pre.cn-hangzhou.aliyuncs.com',
            'timeout_ms': 60000
        }
        agent_bay = AgentBay(api_key, config)
        print("✓ AgentBay client initialized successfully")
        
        # Create VPC session parameters
        params = CreateSessionParams(
            image_id="imgc-07eksy57nw6r759fb",
            is_vpc=True,
            labels={
                "test-type": "vpc-migration-test",
                "purpose": "verify-implementation"
            }
        )
        print("✓ VPC session parameters created")
        
        # Create VPC session
        print("Creating VPC session...")
        session_result = agent_bay.create(params)
        
        if not session_result.success:
            print(f"Error creating session: {session_result.error_message}")
            return False
            
        session = session_result.session
        print(f"✓ VPC session created successfully: {session.session_id}")
        
        # Verify VPC properties
        print(f"✓ Is VPC enabled: {session.is_vpc_enabled()}")
        print(f"✓ Network Interface IP: {session.get_network_interface_ip()}")
        print(f"✓ HTTP Port: {session.get_http_port()}")
        
        # Test MCP tools listing
        print("Testing MCP tools listing...")
        tools_result = session.list_mcp_tools()
        print(f"✓ Found {len(tools_result.tools)} MCP tools")
        
        # Display first few tools
        for i, tool in enumerate(tools_result.tools[:3]):
            print(f"  Tool {i+1}: {tool.name} (server: {tool.server})")
        
        # Test find_server_for_tool functionality
        if tools_result.tools:
            first_tool = tools_result.tools[0]
            found_server = session.find_server_for_tool(first_tool.name)
            print(f"✓ Found server for '{first_tool.name}': {found_server}")
        
        # Clean up
        delete_result = session.delete()
        if delete_result.success:
            print("✓ Session deleted successfully")
        else:
            print(f"Warning: Error deleting session: {delete_result.error_message}")
        
        print("\n🎉 VPC migration test completed successfully!")
        return True
        
    except Exception as e:
        print(f"Error during VPC test: {e}")
        return False

if __name__ == "__main__":
    print("Testing VPC functionality migration...")
    success = test_vpc_session_creation()
    exit(0 if success else 1) 

File: test_vpc_migration.ts
#!/usr/bin/env node
/**
 * Simple test script to verify VPC functionality migration works correctly.
 */

import { AgentBay, CreateSessionParams } from './typescript/src';

async function testVpcSessionCreation(): Promise<boolean> {
    /**Test VPC session creation and tool listing.*/
    // Get API key from environment
    const apiKey = process.env.AGENTBAY_API_KEY;
    if (!apiKey) {
        console.log("Error: AGENTBAY_API_KEY environment variable not set");
        return false;
    }
    
    try {
        // Initialize AgentBay client
        const config = {
            region_id: 'cn-hangzhou',
            endpoint: 'wuyingai-pre.cn-hangzhou.aliyuncs.com',
            timeout_ms: 60000
        };
        const agentBay = new AgentBay({ apiKey, config });
        console.log("✓ AgentBay client initialized successfully");
        
        // Create VPC session parameters
        const params: CreateSessionParams = {
            imageId: "imgc-07eksy57nw6r759fb",
            isVpc: true,
            labels: {
                "test-type": "vpc-migration-test",
                "purpose": "verify-implementation"
            }
        };
        console.log("✓ VPC session parameters created");
        
        // Create VPC session
        console.log("Creating VPC session...");
        const sessionResult = await agentBay.create(params);
        
        if (!sessionResult.success) {
            console.log(`Error creating session: ${sessionResult.errorMessage}`);
            return false;
        }
            
        const session = sessionResult.session!;
        console.log(`✓ VPC session created successfully: ${session.getSessionId()}`);
        
        // Verify VPC properties
        console.log(`✓ Is VPC enabled: ${session.isVpcEnabled()}`);
        console.log(`✓ Network Interface IP: ${session.getNetworkInterfaceIp()}`);
        console.log(`✓ HTTP Port: ${session.getHttpPort()}`);
        
        // Test MCP tools listing
        console.log("Testing MCP tools listing...");
        const toolsResult = await session.listMcpTools();
        console.log(`✓ Found ${toolsResult.tools.length} MCP tools`);
        
        // Display first few tools
        for (let i = 0; i < Math.min(3, toolsResult.tools.length); i++) {
            const tool = toolsResult.tools[i];
            console.log(`  Tool ${i+1}: ${tool.name} (server: ${tool.server})`);
        }
        
        // Test find_server_for_tool functionality
        if (toolsResult.tools.length > 0) {
            const firstTool = toolsResult.tools[0];
            const foundServer = session.findServerForTool(firstTool.name);
            console.log(`✓ Found server for '${firstTool.name}': ${foundServer}`);
        }
        
        // Clean up
        const deleteResult = await session.delete();
        if (deleteResult.success) {
            console.log("✓ Session deleted successfully");
        } else {
            console.log(`Warning: Error deleting session: ${deleteResult.errorMessage}`);
        }
        
        console.log("\n🎉 VPC migration test completed successfully!");
        return true;
        
    } catch (error) {
        console.log(`Error during VPC test: ${error}`);
        return false;
    }
}

async function main() {
    console.log("Testing VPC functionality migration...");
    const success = await testVpcSessionCreation();
    process.exit(success ? 0 : 1);
}

if (require.main === module) {
    main();
} 